openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -115,30 +117,43 @@ public String getPatientIdentifierTypeUuidByName(Config config, String identifie
     }
 
     @Override
+    public List<Identifier> getPatientIdentifierList(Config config, String patientUuid) {
+        String responseJson = getJson(config, "/patient/{patientUuid}/identifier?v=full", patientUuid);
+        return ((IdentifierListResult) JsonUtils.readJson(responseJson, IdentifierListResult.class)).getResults();
+    }
+
+    @Override
     public void deletePatient(Config config, String uuid) {
         delete(config, "/patient/{uuid}?purge", uuid);
     }
 
     @Override
     public void updatePatientMotechId(Config config, String patientUuid, String newMotechId) {
-        Identifier patientIdentifier = getPatientIdentifier(config, patientUuid);
-        patientIdentifier.setIdentifier(newMotechId);
+        List<Identifier> patientIdentifiers = getPatientIdentifierList(config, patientUuid);
+        for (Identifier patientIdentifier : patientIdentifiers) {
+            if ("MOTECH Id".equals(patientIdentifier.getIdentifierType().getName())) {
```

Comment:
this identifier name is configurable. We've got a method to get this name. Please use this here